nasa/ow_autonomy
# PLEXIL uses CameraCapture action

Summary:

- PLEXIL now uses the ROS action for CameraCapture instead of topics (using old name TakePicture).
- All PLEXIL plans containing TakePicture now call CameraCapture, and there there is a new test plan for this action.
- The previous implementation is completely removed.

Unrelated changes:
 - Updated/added comments
 
Test:

- NOTE: this branch must be rebuilt clean:
`$ rm -r <ow-workspace>/build/ow_plexil`
`$ catkin build ow_plexil`
- A clean build itself verifies many of the changes.
- Start any simulator world.
- Run the `TestCameraCapture` plan, e.g.
`$ roslaunch ow_plexil ow_exec.launch plan:=TestCameraCapture.plx`
- Watch both Gazebo and the ow_exec terminal, which explains what you should see in Gazebo as the plan runs.
- Run the plan TestAntennaCamera for regression.  It should complete without incident.


Notes:
 - There is a known bug, unrelated to this work, that will manifest in any plan that takes images.  The following sequence will appear at intervals.
   ` [WARN]: Fault in SYSTEM: CAMERA_EXECUTION_ERROR`
    `[WARN]: Resolved fault in SYSTEM: CAMERA_EXECUTION_ERROR`
 - There is a caveat in the test above, in that a failed image capture will leave a stale image in RQT.  One solution would be to move the antenna a bit between images, though this would make visually detecting exposure changes harder since two things instead of one could change between tests.  The best solution would be making this test non-interactive and entirely programmatic and automatic.
